### DISCUSSION TITLE
Debian 10 Buster: Web Update not working

### DISCUSSION THREAD

**Discussion by D3nt0n1418** [FreshRSS/FreshRSS#6915] [Q&A] [answered]
### Describe the bug

Attempting to update 1.24.2-dev to latest version. After clicking "Start update" the screen goes white and I'm unable to access FreshRSS. Was forced to revert the VM to an older snapshot. I've been able to update via this method in the past.

### To Reproduce

1. Go to Update page
2. Click "Check for new updates"
3. Click "Start update"

### FreshRSS version

1.24.2-dev

### Environment information

- Database version: [e.g. Mysql 5.7, SQLite]
mysql  Ver 15.1 Distrib 10.5.23-MariaDB, for debian-linux-gnu (x86_64) using  EditLine wrapper
- PHP version: [e.g. PHP 8.1]
7.4.33
- Installation type: [e.g. Docker, Docker image source, git, Yunohost]
Installed on a VM, I think from git but it's been awhile since I've set this up
-Web server type: [e.g. Apache, nginx]
Apache
- Device: [e.g. iPhone13]
Laptop
- OS: [e.g. Ubuntu 22.04, Win10, MacOS14]
Debian 10.2.1-6
- Browser: [e.g. Firefox 124]
Firefox

**Comment by Alkarex** [comment, marked as answer]
The likely problem is that we do not support PHP 7 anymore (minimum is PHP 8.1+). 

See:
* https://github.com/FreshRSS/FreshRSS/releases/tag/1.24.3 (last version supporting PHP 7.4)
* https://github.com/FreshRSS/FreshRSS/blob/edge/CHANGELOG.md
* https://github.com/FreshRSS/FreshRSS/blob/edge/README.md#requirements

But you are supposed to get something in your Web server logs.
Reverting to an old VM snapshot is overkill: if installed by git, you can just git reset to the previous FreshRSS version, in this case 1.24.3.

Solutions include:
* Updating to Debian 12 Bookworm, which [ships with PHP 8.2](https://packages.debian.org/bookworm/php)
* Use our [Docker image](https://hub.docker.com/r/freshrss/freshrss/) `freshrss/freshrss:edge`
* Install PHP 8.1+ on your existing Debian 10, e.g. via a package from https://deb.sury.org/
* Stay a bit longer on FreshRSS 1.24.3

**Reply by D3nt0n1418** [reply]
You were correct. Updating PHP fixed the issue and I am now on the latest version.
Thank you!

**Reply by Frenzie** [reply]
Also see https://debian.octopuce.fr/snapshots/sury-php/20240710010404/ (the last one that lists buster)
